web/src/components/graph/areagraph.js

```diff
@@ -0,0 +1,132 @@
+import React, {
+  useState,
+  useEffect,
+  useRef,
+  useMemo,
+} from 'react';
+import { scaleTime, scaleLinear } from 'd3-scale';
+import { first, last } from 'lodash';
+import moment from 'moment';
+
+import AreaGraphGradients from './areagraphgradients';
+import AreaGraphLayers from './areagraphlayers';
+import GraphBackground from './graphbackground';
+import GraphHoverLine from './graphhoverline';
+import ValueAxis from './valueaxis';
+import TimeAxis from './timeaxis';
+
+const X_AXIS_HEIGHT = 20;
+const Y_AXIS_WIDTH = 35;
+const Y_AXIS_PADDING = 4;
+
+const getTimeScale = (containerWidth, datetimes, currentTime) => scaleTime()
+  .domain([first(datetimes), currentTime ? moment(currentTime).toDate() : last(datetimes)])
+  .range([0, containerWidth]);
+
+const getValueScale = (containerHeight, maxTotalValue) => scaleLinear()
+  .domain([0, maxTotalValue * 1.1])
+  .range([containerHeight, Y_AXIS_PADDING]);
+
+const AreaGraph = React.memo(({
+  id,
+  layers,
+  currentTime,
+  maxTotalValue,
+  valueAxisLabel,
+  datetimes,
+  fillSelector,
```

Comment:
what is this?

Reply:
It was a function for selecting layer fill color based on layer index. I simplified this in https://github.com/tmrowco/electricitymap-contrib/pull/2171/commits/edae3eccca2dc79a1d3ad013cea9dcb6c6fcb498 so that the fill colors are attached to the `layers` prop instead of being passed around as a separate prop.